aiohttp/web_urldispatcher.py

```diff
@@ -91,53 +84,113 @@ def _append_query(url, query):
             return url
 
 
-class PlainRoute(Route):
+class ViewableRoute(Route):
+
+    def __init__(self, name, *, expect_handler=None):
+        super().__init__(name, expect_handler=expect_handler)
+        self._views = []
+
+    def match_view(self, request, view_name, match_dict):
+        method = request.method
+        allowed_methods = set()
+
+        for view in self._views:
+            if (view.name == view_name and
```

Comment:
Another issue with the list choice is the silent collision here. Assume the following (pseudo)-code:

```
route = app.add_route('/foo/bar')
route.add_view(METH_ANY, handle_any)
route.add_view(METH_GET, handle_get)
```

The idea behind is to catch GET requests with the special hander, but everything else with some generic one. But here it wouldn't work because METH_ANY would have higher resolution order (which depended on declaration one) while technically, it should be used as fallback case when no direct method match is possible.